app/frontend/actions.ts

```diff
@@ -164,20 +165,26 @@ export default (store: Store) => {
 
   /* TRANSACTION */
 
-  const confirmTransaction = async (state: State, txConfirmType): Promise<void> => {
+  const _confirmTransaction = async (
+    state: State,
+    {
+      txConfirmType,
+      txPlan,
+      sourceAccountIndex,
+    }: {txConfirmType: TxType; txPlan: TxPlan; sourceAccountIndex: number}
+  ): Promise<void> => {
     let txAux
-    const newState = getState()
     try {
-      if (newState.sendTransactionSummary.plan) {
+      if (txPlan) {
```

Comment:
FYI: basically what was changed is that the `state.something` was exchanged for explicit params